### PR TITLE
Error handling headers

### PR DESCRIPTION
**Pull request by SwampApe**
I made the changes described in #2602 and added tests. I'm fairly new to this project so let me know if there is anything I need to change/fix.
#### Checklist
- [ ] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/error-handling-headers.test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+const t = require('tap')
```
Comment (jsumners) [inline test/error-handling-headers.test.js:1]:
```suggestion
'use strict'

const t = require('tap')
```

Do we need a new file? Where is this code already tested? Can this test not live there?

Comment (zekth) [reply test/error-handling-headers.test.js:1]: why not `reply.test.js`?

Comment (SwampApe) [reply test/error-handling-headers.test.js]: Just moved the tests to reply-error.test.js instead.

**Comment by mcollina** [comment]
linting is failing (see CI)

**Comment by SwampApe** [comment]
Seems like some of the old tests are expecting the error headers after setting a custom error handler. Thoughts on what I should do?

**Comment by mcollina** [comment]
Go ahead and fix them. This PR might end up being semver-major.

**Pull request closed without merging**